iati/tests/test_resources.py

```diff
@@ -50,193 +177,492 @@ class TestResourceFolders(object):
 
-        assert len(activity_paths) == 1
+        assert len(result) == 1
+        assert result[0] == iati.resources.create_codelist_mapping_path(std_ver_minor_mixedinst_valid_fullsupport)
 
-    def test_get_organisation_schema_paths(self, std_ver_minor_mixedinst_valid_fullsupport):
-        """Check that all organisation schema paths are found.
+    def test_get_codelist_mapping_paths_independent(self):
+        """Test getting a list of version-independent Codelist Mapping files."""
+        result = iati.resources.get_codelist_mapping_paths(iati.version.STANDARD_VERSION_ANY)
 
-        Todo:
-            Handle all paths to schemas being found correctly.
+        assert result == []
 
-        """
-        organisation_paths = iati.resources.get_organisation_schema_paths(std_ver_minor_mixedinst_valid_fullsupport)
+    def test_get_codelist_mapping_paths_minor_partsupport(self, std_ver_minor_mixedinst_valid_partsupport):
+        """Test getting a list of Codelist Mapping paths. The requested version is partially supported by pyIATI."""
+        result = iati.resources.get_codelist_mapping_paths(std_ver_minor_mixedinst_valid_partsupport)
 
-        assert len(organisation_paths) == 1
+        assert result == []
 
-    def test_find_schema_paths(self, std_ver_minor_mixedinst_valid_fullsupport):
-        """Check that both the activity and organisation schema paths are being found.
+    def test_get_codelist_mapping_paths_minor_unknown(self, std_ver_all_mixedinst_valid_unknown):
+        """Test getting a list of Codelist Mapping paths. The requested version is not known by pyIATI."""
+        result = iati.resources.get_codelist_mapping_paths(std_ver_all_mixedinst_valid_unknown)
 
-        Todo:
-            Handle all paths to schemas being found correctly.
+        assert result == []
 
-        """
-        paths = iati.resources.get_all_schema_paths(std_ver_minor_mixedinst_valid_fullsupport)
+    def test_get_codelist_mapping_paths_major_known(self, std_ver_major_uninst_valid_known):
+        """Test getting a list of Codelist Mapping paths. The requested version is a known integer version. The list should contain paths for each supported minor within the major."""
+        supported_versions_at_major = [version for version in iati.version.versions_for_integer(std_ver_major_uninst_valid_known) if version in iati.version.STANDARD_VERSIONS_SUPPORTED]
+        expected_path_count = len(supported_versions_at_major)
+
+        result = iati.resources.get_codelist_mapping_paths(std_ver_major_uninst_valid_known)
+
+        assert len(result) == expected_path_count
+        for version in supported_versions_at_major:
+            assert iati.resources.create_codelist_mapping_path(version) in result
 
-        assert len(paths) == 2
 
-    @pytest.mark.parametrize('create_schema_path_function', [
+@pytest.mark.new_tests
+class TestResourceGetRulesetPaths(object):
```

Comment:
Add a note in the code (as well as the existing commit message) detailing why this looks very similar to the equivalent class for Codelist Mapping files, but should be separate:

* Different numbers of parameters to the functions being tested
* It's possible that maybe their behaviour may diverge a bit, possibly